instrumentation.go

```diff
@@ -57,6 +58,10 @@ type Instrumentation struct {
 	target   *process.TargetDetails
 	analyzer *process.Analyzer
 	manager  *instrumentation.Manager
+
+	stopMu  sync.Mutex
+	stop    context.CancelFunc
+	stopped chan struct{}
```

Comment:
nit: we could group these 3 fields into a "Closing" struct

Reply:
I looked into this, but am not sure it is worth it. There is a lot of coordination between the setup and tear-down of these within the context of the `Intrumentation` methods that they are pretty tightly coupled. Without doing a more complete re-design on the signaling within the `Instrumentation`, this is probably going to be the best we can do.

I'm not opposed to that re-design. Just, I would want to address it in a separate issue.

Going to resolve this, and track the work in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1006.